File: 0071_Simplify Path/SimplifyPath.cs
public class Solution {
    public string SimplifyPath(string path) {
        if(string.IsNullOrEmpty(path)) return "";
        var stack = new Stack<string>();
        var splits = path.Split('/');
        foreach(var split in splits)
        {
            if(split == string.Empty || split.Equals(".")) continue;
            else if(split.Equals(".."))
            {
                if(stack.Count > 0) stack.Pop();
            }else{
                stack.Push(split);
            }
        }
        
        var sb = new StringBuilder();
        var list = stack.ToList();
        list.Reverse();
        foreach(var s in list)
        {
            sb.Append("/");
            sb.Append(s);
        }
        
        var ans = sb.ToString();
        return string.IsNullOrEmpty(ans) ? "/" : ans;
        
    }
}

File: 0071_Simplify Path/SimplifyPath.java
class Solution {
    public String simplifyPath(String path) {
        if(path.isBlank()) return "/";
        String[] strs = path.split("/");
        Stack<String> stack = new Stack<>();
        for(String s : strs) {
            if(s.isEmpty() || s.equals(".")) continue;
            
            if(s.equals("..")) {
                if(stack.size() > 0) stack.pop();
            }else {
                stack.push(s);
            }
        }
        
        if(stack.size() == 0) return "/";
        
        StringBuilder sb = new StringBuilder();
        while(stack.size() > 0){
            sb.insert(0, stack.pop());
            sb.insert(0, "/");
        }
        
        return sb.toString();
    }
}